packages/api/tests/endpoints/providers/delete-provider.js

```diff
@@ -109,25 +137,82 @@ test.todo('Attempting to delete a provider with an unauthorized user returns an
 
 test('Deleting a provider removes the provider', async (t) => {
   const { testProvider } = t.context;
-
+  const id = testProvider.id;
   await request(app)
-    .delete(`/providers/${testProvider.id}`)
+    .delete(`/providers/${id}`)
     .set('Accept', 'application/json')
     .set('Authorization', `Bearer ${jwtAuthToken}`)
     .expect(200);
 
+  const rdsRecord = await t.context.testKnex(tableNames.providers).select().where({
+    name: id,
+  });
+
   t.false(await providerModel.exists(testProvider.id));
+  t.is(rdsRecord.length, 0);
 });
 
 test('Deleting a provider that does not exist succeeds', async (t) => {
   const { status } = await request(app)
     .delete(`/providers/${randomString}`)
     .set('Accept', 'application/json')
     .set('Authorization', `Bearer ${jwtAuthToken}`);
-
   t.is(status, 200);
 });
 
+test('Attempting to delete a provider with an associated postgres rule returns a 409 response', async (t) => {
+  const { testProvider } = t.context;
+  const rule = fakeRuleFactoryV2({
+    provider: testProvider.id,
+    rule: {
+      type: 'onetime',
+    },
+  });
+
+  // The workflow message template must exist in S3 before the rule can be created
+  await s3().putObject({
+    Bucket: process.env.system_bucket,
+    Key: `${process.env.stackName}/workflows/${rule.workflow}.json`,
+    Body: JSON.stringify({}),
+  }).promise();
+
+  await ruleModel.create(rule);
+
+  // This block will need to be refactored as the Collection and Rule endpoints
+  // are updated on the feature branch
+  // ---
+  const providerResult = await t.context.testKnex(tableNames.providers)
+    .select('cumulusId')
+    .where('name', testProvider.id);
+
+  const collectionResult = await t.context.testKnex(tableNames.collections).insert({
+    name: randomString(10),
+    version: '001',
+    sampleFileName: 'fake',
+    granuleIdValidationRegex: 'fake',
+    granuleIdExtractionRegex: 'fake',
+    files: {},
+  }).returning('cumulusId');
+
+  await t.context.testKnex(tableNames.rules).insert({
```

Comment:
To validate this behavior, we need to have rule/collection inserts here -- and given we don't have an endpoint to hit to add them (yet), this seems workable.